core/asset/service.go

```diff
@@ -80,12 +81,25 @@ func (s *Service) DeleteAsset(ctx context.Context, id string) error {
 	return s.discoveryRepository.DeleteByURN(ctx, id)
 }
 
-func (s *Service) GetAssetByID(ctx context.Context, id string) (Asset, error) {
+func (s *Service) GetAssetByID(ctx context.Context, id string) (ast Asset, err error) {
 	if isValidUUID(id) {
-		return s.assetRepository.GetByID(ctx, id)
+		if ast, err = s.assetRepository.GetByID(ctx, id); err != nil {
+			return ast, fmt.Errorf("error when getting asset by id: %w", err)
```

Comment:
Sure no issue here, but just wondering what would be concern of returning `ast`?

CMIIW if error is not nil, ideally the returned asset won't matter right?

Reply:
We should always set the expectation that the value returned with an error should be discarded. Doing otherwise can lead to incorrect usage of API. For ex: Returning a possibly valid asset without probes when call to fetch probes fails.